### ISSUE TITLE
Release 5.0.0

### ISSUE THREAD

**Issue by pascalwengerter** [owncloud/web#6248]
## Waiting for
- #6156
- #6199
- #6262
- #6213
- #6228
- #6240
- https://github.com/owncloud/owncloud-design-system/issues/1932
- https://github.com/owncloud/web/issues/6369
- https://github.com/owncloud/web/pull/6287/
- https://github.com/owncloud/web/pull/6331
- https://github.com/owncloud/web/pull/6381
- https://github.com/owncloud/web/issues/6326
- SDK official 2.0 release & https://github.com/owncloud/owncloud-sdk/pull/979
- https://github.com/owncloud/web/pull/6386
- https://github.com/owncloud/web/pull/6401
- https://github.com/owncloud/web/pull/6399
- https://github.com/owncloud/web/pull/6403
- https://github.com/owncloud/web/pull/6418

## If time permits
- https://github.com/owncloud/web/issues/6383
- https://github.com/owncloud/web/issues/6280

## Small optional fixes / wishes (maybe on release branch)
- [ ] checkboxes have no focus border anymore
- [X] left sidebar toggle "expand" is not visible anymore (collapsed mode) https://github.com/owncloud/web/pull/6386
- [X] left sidebar nav items have white padding on the right (collapsed mode) https://github.com/owncloud/web/pull/6386

## Release checklist
- [x] pin release issue @pascalwengerter 
- [x] prepare release rc: @kulmann
  - [x] create a branch `release-5.0.0` @kulmann
  - [x] update the version in `packages/web-integration-oc10/appinfo/info.xml` @kulmann
  - [x] update the version in `package.json` @kulmann
  - [x] update the version in `sonar-project.properties` @kulmann 
  - [x] move changelog/unreleased/* to changelog/VERSION_YYYY-MM-DD @kulmann
  - [x] update the ocis commit id in `.drone.env` to the most recent one from ocis master (let's see how it goes) @kulmann
  - [x] commit, push and create **draft** PR (#...) for this branch @kulmann
- [x] create pre release tag `v5.0.0-rc` from the release branch @kulmann 
- [x] smoke test 
  - [x] against oCIS @pascalwengerter  @ScharfViktor 
  - [x] against oc10 standalone @kulmann 
  - [x] against oc10 as app installation @kulmann 
  - [x] against oCIS wopi example / app provider @pascalwengerter 
- [x] if needed, fix issues or write down known issues 
- [x] update `YYYY-MM-DD` to current date in `changelog/VERSION_YYYY-MM-DD`
- [x] Make sure translations are up to date by updating them in Transifex and restarting translation sync job https://drone.owncloud.com/owncloud/translation-sync/ @pascalwengerter
- repeat above steps if a new rc is needed
- [x] Run translations sync job once again
- [x] go to the releases section and click `Draft a new Release`. Use `v5.0.0` as a tag (the v prefix is important) and `5.0.0` as name and publish it
- [x] ping in internal #btr channel to create a signed release and ask them to upload it to a) the github release page, b) marketplace.owncloud.com, c) portal.owncloud.com, d) customer.owncloud.com
- [ ] merge release branch https://github.com/owncloud/web/pull/... back to master (CI must be green)

**Comment by dschmidt** [comment]
I would really like to see https://github.com/owncloud/web/pull/6156 getting into the release. I made sure to make it readytoreview before my vacation because file/folder handling inconsistencies in different apps have superannoying consequences like certain apps only working in certain contexts... and we need this pr to properly fix it for all apps

**Comment by dschmidt** [comment]
https://github.com/owncloud/core/issues/39707

**Comment by pascalwengerter** [comment]
Smoke testing oCIS 1.16 with the WOPI deployment example (edge has broken the WOPI integration AFAIK) and the v5.0.0  leads to certification errors (we used to prevent them by visiting the editor URL, e.g. codimd.owncloud.test, and choosing the "accept the risk and continue" button  before opening the editor, which doesn't seem to work anymore)

Screenshots (Chrome & FF):
<img width="1440" alt="Screenshot 2022-02-11 at 13 45 19" src="https://user-images.githubusercontent.com/16822008/153593903-528775ce-582b-4a44-8605-82093fc8d46f.png">
![Screenshot 2022-02-11 at 13-45-34 External app - ownCloud](https://user-images.githubusercontent.com/16822008/153593919-8846cb96-95ea-46e6-8670-28421d0b1b0f.png)

**Comment by ScharfViktor** [comment]
#  SmokeTest  against ocis
branch: v5.0.0-rc

### Case 1: admin share folder(with file) to marie:

Admin:
- [ ]  upload big file 30gb - very slow 
- [ ]  upload big file 10gb - very slow
- [x] upload folder with 848 files. 
- [x] Check pagination, count items per page. 
- [ ] upload folder(drag folder) "folder_to_share" with file.pdf Bug: https://github.com/owncloud/web/issues/6426
- [x] share folder "folder_to_share" to marie with edit right

Marie:
- [x] accept "folder_to_share"
    ❌ - when you try to rename file before accept you get 404 Error. Which I think is right. Do we need button "rename"? 
![image](https://user-images.githubusercontent.com/84779829/153603546-db0ffdb1-07a3-4e25-97d4-8512460e0873.png)


- [ ] rename "folder_to_share" (?admin don't see)
- [ ] upload new file "file.txt" to "folder_to_share" (admin see new file)
- [ ] open pdf file
- [ ] rename file.pdf (use rigth click) (admin see that file changed name)
- [ ] try to move "folder_to_share" to "/" (400)
- [ ] copy "folder_to_share" to "/"

Admin:
- [ ] add new file.jpeg to "folder_to_share" (marie can see file without access)
- [ ] add file format drawio(marie can open in Draw.io and can edit) 
- [ ] move file.pdf to "/" (marie can not see file.pdf)
- [ ] add new "file.txt" create new version. (marie can see new version)
- [ ] open txt file in markdown-editor. Change and save.  ❌ created a file with UUID name in "All files"

Marie:
- [ ] can download both versions (!error 501)
- [ ] restore old version

Admin:
- [ ] change share "folder_to_share" from "edit" to "viewer" (marie cannot rename/move/delete/download file in folder and cannot create new document in the folder)

Marie:
- [ ] decline folder (but folder exist in Shares - bug) - existing bug

Admin:
- [ ] delete folder "folder_to_share" (marie: folder Shares empty)

## Case 2: admin share file to marie:

Admin:
- [ ] upload file.jpeg
- [ ] share file.jpeg to marie with rigth viewer (use main menu. find marie by email marie@example.org)

Marie:
- [ ] accept file (marie can open/download/copy file but cannot delete/move/rename)
- [ ] copy file to "/"
- [ ] open file in Mediaviewer
- [ ] download file

Admin:
- [ ] change share "file.jpeg" from "viewer" to "edit"(marie can open/download/copy/rename file)

Marie:
- [ ] rename file (admin don't see)

Admin:
- [ ] unshare "file.jpeg" (marie cannot see "file.jpeg" in Shares/Shared for me but see the copied file)

## Case 3: admin create public links to folder

Admin:
- [ ] create folder "for upload"
- [ ] add link with name "to_upload" with role "Uploader" with pass and expiration date
- [ ] open link (can only upload)
- [ ] upload file (admin can see new file)
- [ ] refresh page(page is empty)
- [ ] change share "for upload" from "Uploader" to "edit" 
- [ ] open old link (it work as upload)
- [ ] open new link (can open/download/copy/rename/move files)
- [ ] admin add new file to "for upload"
- [ ] refresh link (file exist)
- [ ] admin change pass (refresh link)

## Case 4: admin create link to file

Admin:
- [ ] create file png
- [ ] add link with main menu
- [ ] open link(can open, download, see) in the incognito tab

- [ ] change link, add pass
- [ ] refresh page (input pass)
- [ ] delete link
- [ ] refresh page (resource not found)
- [ ] delete file
- [ ] restore
- [ ] delete
- [ ] Empty trash

## Case 5: create new user:

Admin:
- [ ] create new account "tom"

Tom:
- [ ] go to account
- [ ] change language

Admin:
- [ ] block user (Login Error access denied)
- [ ] delete user (Logon failed)
- [ ] logout

**Comment by kulmann** [comment]
## Smoke test v5.0.0-rc against oc10 app

- :heavy_check_mark: as gunter - upload arch/mips folder from linux kernel (1945 files)
        - ❌ drag and drop folder upload completely broken 👎 error `the server responded with a status of 403 (Forbidden)`
        - via folder upload button works
	- progress bar is not sticky
	- expanding the upload progress section slows down browser & upload time significantly
- :heavy_check_mark: as gunter - share the folder to alice as viewer
        - sharing is not possible before the upload finished (user invite field doesn't stay focussed) 
        - ❌ for reaching the invite form field one needs to tab through the whole file list until the sidebar is reached (focus should be set to sidebar or sidebar panel instead, depending on a11y needs)
- :heavy_check_mark: as alice - share is already accepted. declining and accepting again works as expected (buttons in shared with me table, batch action, right sidebar)
- :heavy_check_mark: as alice - try do delete file from shared folder
        - individual delete action not visible, good
        - batch delete action not visible, good
        - context menu action not visible, good
- :heavy_check_mark: as alice - try to add file
        - not possible (which is expected) but drag & drop upload didn't handle it. opened file in new tab instead.
- ❌ as alice - copy 16 folders from share to home
	- no visible feedback that it's in progress
	- UI still weirdly usable, ended up in an active upload drop zone. Needed a page reload to be usable again.
	- folder listing after copy shows `?` as folder size on some of the copied folders.
- :heavy_check_mark: as gunter - add new file to share
- :heavy_check_mark: as alice - receive new file in existing share
- :heavy_check_mark: as gunter - unshare folder from shared with others 
- :heavy_check_mark: as alice - try to revisit old share
	- shows `resource not found` upon page reload, good
	- going back to `All files` or `Shared with me` doesn't list the share anymore, good
- :heavy_check_mark: as gunter - share folder with alice again, but as editor
        - setting and removing an expiration date work as expected, good
- :heavy_check_mark: as alice - rename file
	- via right sidebar, works, good
	- via context menu, works, good
- :heavy_check_mark: as gunter - add file
- :heavy_check_mark: as alice - delete file
	- via right sidebar, works, good
	- via context menu, works, good
	- via batch action, works, good
- :heavy_check_mark: as alice - create public link on any folder
- :heavy_check_mark: as alice delete file and recover it
- :heavy_check_mark: as alice create version of textfile
- :heavy_check_mark: as alice view version history of text file
- :heavy_check_mark: as alice download newest version of text file
- :heavy_check_mark: download older version of text file
- :heavy_check_mark: restore older version of text file
	- no visible feedback about the success
	- preview is not updated

- :heavy_check_mark: as alice - upload avatar in classic UI
	- see own avatar in top bar (far right) 
	- login as gunter, having a share with alice
		- `shared with others` page loads and shows avatars
	- avatar in top bar doesn't have correct aspect ratio, see screenshot
<img width="163" alt="Screenshot 2022-02-11 at 15 51 55" src="https://user-images.githubusercontent.com/3532843/153613682-ee2f63d8-ba96-4865-a3ad-6ead51bbe731.png">

- :heavy_check_mark: as alice, move a file into subfolder via drag and drop
- :heavy_check_mark: as alice, move multiple files & folders into subfolder via drag and drop
- :heavy_check_mark: as alice, navigate into a folder, open right sidebar without selection, share current folder with gunter

### Findings from old smoke testing:
- when a share fails, the submit button of the form stays in the disabled state with spinner. Needs to reset itself instead. (not resolved yet!)

### Findings from new smoke testing:
- avatar in top bar doesn't have correct aspect ratio, see screenshot
<img width="163" alt="Screenshot 2022-02-11 at 15 51 55" src="https://user-images.githubusercontent.com/3532843/153613682-ee2f63d8-ba96-4865-a3ad-6ead51bbe731.png">
- drag and drop folder upload completely broken

**Comment by ChrisEdS** [comment]
Settings menu placement looks wrong:

![image](https://user-images.githubusercontent.com/33026403/153626299-d5b4033c-c502-4d0e-9db7-20868e5d90bc.png)

Reduce at least the spacing to the gear icon?

**Comment by pascalwengerter** [comment]
> Settings menu placement looks wrong:
> 
> ![image](https://user-images.githubusercontent.com/33026403/153626299-d5b4033c-c502-4d0e-9db7-20868e5d90bc.png)
> 
> Reduce at least the spacing to the gear icon?

@kulmann that was introduced by giving the `.tippy-content` additional padding to make the shadow visible 🥴

**Comment by pascalwengerter** [comment]
I have a fix for the avatar image width/height mess-up. Push straight to release branch?

**Comment by kulmann** [comment]
> I have a fix for the avatar image width/height mess-up. Push straight to release branch?

Yep, push it :-)

**Comment by ChrisEdS** [comment]
Low contrast in folder action menu

<img width="367" alt="image" src="https://user-images.githubusercontent.com/33026403/153775320-55db8229-5b35-4e37-8ab4-0208a853ea92.png">

**Comment by ChrisEdS** [comment]
Buttons are to big in my opinion:

![image](https://user-images.githubusercontent.com/33026403/153831497-6c37bd08-2e12-4d3f-960e-78719d0307ab.png)

**Comment by ScharfViktor** [comment]
> Buttons are to big in my opinion:
> 
> ![image](https://user-images.githubusercontent.com/33026403/153831497-6c37bd08-2e12-4d3f-960e-78719d0307ab.png)

Do we have access to the layout (e.g. figma) of the new design? 
to see if it's a mistake or if it's meant to be

**Comment by kulmann** [comment]
There's a fine line between "UI/UX feedback for future improvement" and "glitches that need to be fixed in this release". @ChrisEdS could you create a separate issue from your comment about the batch action button size? Won't fix it in this release, because there's more UX discussion to it than just shrinking the size.

**Comment by kulmann** [comment]
Dark mode search preview has the same background color and parent-folder link color, see screenshot

<img width="463" alt="Screenshot 2022-02-14 at 16 08 04" src="https://user-images.githubusercontent.com/3532843/153890088-4c65c94e-e6a4-46d3-badf-3fe2744905a4.png">

**Comment by pascalwengerter** [comment]
#  SmokeTest  against oCIS (binary @ current master)
branch: `release-5.0.0`

### Case 1: admin share folder(with file) to marie:

einstein:
- [x]  upload big file 30gb - very slow 
- [x]  upload big file 6.6gb - reasonably fast
- [x] upload folder with 848 files. 
- [x] Check pagination, count items per page. 
- [x] upload folder via drag&drop (7.111 files) - progress bar doesn't move while count goes from 1 to 7.111, then does track progress while counting down again. Upload interupted by OIDC refresh
- [x] share folder "101_archive to marie with edit right

Marie:
- [x] accept "101_archive"
- [x] ❌ when trying to rename file before accept you get 404 Error. => we should not show the rename action on pending shares (folders)
- [x] rename "101_archive" to "archive_2" works for Marie, but doesn't show renamed folder for einstein
- [x] upload new file "file.txt" to "101_archive", is visible for einstein
- [x] open pdf file
- [x] rename file in "archive_2", einstein sees that file has been renamed
- [x] try to move "101_archive" to "user home" (leads to error message)
- [x] copy "101_archive" to "user home" (no visible feedback for copying this many files, browser session is totally wrecked after trying to navigate into copied folder which has only partly copied files)

Einstein:
- [x] add new file.jpeg to "101_archive" (marie can see file)
- [x] add file format drawio (marie can open in Draw.io and can edit) 
- [x] move file.pdf to "/" (marie can not see file.pdf)
- [x] add new "file.txt" create new version. (marie can see new version)
- [x] open txt file in markdown-editor. Change and save.

Marie:
- [x] ❌ can download both versions (!error 501)
- [x] ❌ restore old version toggles between two older states

Einstein:
- [x] change share "101_archive" from "edit" to "viewer" (marie cannot rename/move/delete/download file in folder and cannot create new document in the folder)

Marie:
- [x] decline folder 

Einstein:
- [x] delete folder "folder_to_share" (marie: folder Shares empty)

## Case 2: admin share file to marie:

Einstein:
- [x] upload file.jpeg
- [x] share file.jpeg to marie with viewer permissions

Marie:
- [x] accept file (marie can open/download/copy file but cannot delete/move/rename)
- [x] copy file to "/"
- [x] open file in Mediaviewer
- [x] download file

Einstein:
- [x] rename "file.jpeg" to "file2.jpeg", marie doesn't see renamed share but original filename "file.jpeg"
- [x] change share permissions for "file2.jpeg" from "viewer" to "edit"(marie can open/download/copy/rename file)

Marie:
- [x] rename file to "file3.jpg", einstein doesn't see renamed share but original filename "file2.jpeg"

Einstein:
- [x] unshare "file.jpeg" (marie cannot see "file.jpeg" in Shares/Shared for me but see the copied file)

## Case 3: einstein create public links to folder

einstein:
- [x] create folder "for upload"
- [x] add link with name "to_upload" with role "Uploader" with password and expiration date
- [x] open link (can only upload)
- [x] upload file (einstein can see new file)
- [x] refresh page(page is empty)
- [x] change share "for upload" from "Uploader" to "edit"
- [x] open link again (can open/download/copy/rename/move files)
- [x] einstein add new file to "for upload"
- [x] refresh link (file exist)
- [x] einstein change password (refresh link)

## Case 4: einstein create link to file

einstein:
- [x] create file png
- [x] create view permission link share in sidebar
- [x] open link in the incognito tab
- [x] einstein changes link by adding a password
- [x] incognito refresh page, inputs password, sees file
- [x] einstein deletes link
- [x] incognito refresh page, sees only "resource not found"
- [x] einstein deletes file
- [x] einstein restores file
- [x] einstein deletes file
- [x] einstein empties trash

## Case 5: create new user:

Admin:
- [ ] ❌ create new account "tom"

Tom:
- [ ] ❌ go to account https://github.com/owncloud/web/issues/6430
- [ ] ❌ change language

Admin:
- [ ] ❌ block user (Login Error access denied)
- [ ] ❌ delete user (Logon failed)
- [ ] ❌ logout

## Findings: 
- The "Accounts" tab is not available https://github.com/owncloud/web/issues/6430
- ~Change role in public link doesn't work https://github.com/owncloud/web/issues/6429~
- Archiver doesn't work in public link https://github.com/owncloud/web/issues/6423 (no capabilities on password protected public link, archiver returns 404 on non-password protected public link)
- ~Upload folder via drag and drop is broken https://github.com/owncloud/web/issues/6426~

**Comment by ChrisEdS** [comment]
Jumping file list content in small resolution

https://user-images.githubusercontent.com/33026403/153903022-81f50d97-7289-40d3-a77c-8ad9599166b0.mov
